### PR TITLE
Helm system repo refactor and fix

### PR DESCRIPTION
**Pull request by sk4zuzu**
This is the second PR in the series. :hand_over_mouth:

When I was checking what needs to be done to make custom repository/image_registry's code working properly I found multiple bugs in our helm code. :disappointed:

This PR does couple of things:
- removes dangerous code that was deleting helm releases from "default" namespace automatically (I don't think we should be doing that) - IMO much better place for it is to do it during upgrades and I recommend that! :+1::innocent:
- mentioned code had also two bugs as well (it means it was invalid), for example this code is incorrect:
```yaml
      failed_when: (helm_charts_list_before_update.rc != 0 and not 'no repositories' in helm_charts_list_before_update.stderr)	
                   or helm_charts_list_before_update.rc > 1
```
```yaml
      shell: helm delete {{ chart_subname }} --purge
``` 
- improves `install-system-chart.yml` task file, fixes bugs and adds dedicated namespace for "system" charts
- adds "dummy" chart for testing purposes (it is difficult to test helm feature without any charts in place)
- removes the confusion we caused by assuming helm release is called just the same as the chart name (`helm delete <chart-name>` is confusing and incorrect)
- updates documentation
- renames `helm-charts` to `charts` in urls to keep it consitent with our repository/offline-package simple conventions

:hugs:

### CODE REVIEW

**Review by erzetpe — DISMISSED** [review]

File: docs/home/howto/HELM.md [hunk 1]
```diff
@@ -1,78 +1,67 @@
...
+Epiphany provides helm repository for internal usage inside our ansible codebase. Currently only the "system" repository is available, but it's not designed to be used by regular users. __In fact, regular users must not reuse it for any purpose.__
 
-System Helm charts are stored in the following location:
+Epiphany developers can find it inside this location `roles/helm_charts/files/system`. To add a chart to the repository it's enough just to put unarchived chart directory tree inside the location (in a separate directory) and re-run `epcli apply`.
 
-`roles/helm_charts/files/system`
+When the `repository` ansible role is run it copies all unarchived charts to the repository host, creates standard helm `index.yaml` and serves all this data from Apache HTTP server.
 
-There is also dedicated Helm charts location for users
+## Installing helm releases from the "system" repository charts
```
Comment (to-bar) [inline docs/home/howto/HELM.md:9]:
```suggestion
## Installing Helm charts from the "system" repository
```

File: docs/home/howto/HELM.md [hunk 2]
```diff
@@ -1,78 +1,67 @@
...
 
-`roles/helm_charts/files/user`
+Epiphany developers can reuse the "system" repository from any place inside the ansible codebase. Moreover, it's a responsibility of a particular role to call the `helm upgrade --install` command.
```
Comment (to-bar) [inline docs/home/howto/HELM.md:11]:
```suggestion
Epiphany developers can reuse the "system" repository from any place inside the Ansible codebase. Moreover, it's a responsibility of a particular role to call the `helm upgrade --install` command.
```

File: docs/home/howto/HELM.md [hunk 3]
```diff
@@ -1,78 +1,67 @@
...
 
-- Currently user part has no automation implemented. It will need some additional "options" via epi client but it needs discussion with wider audience about the needs.
+Of course, there is a hepler task file that can be reused for that purpose `roles/helm/tasks/install-system-release.yml`. __It's only responsible for installing already existing "system" helm charts from the "system" repository.__
```
Comment (to-bar) [inline docs/home/howto/HELM.md:13]:
```suggestion
There is a helpler task file that can be reused for that purpose `roles/helm/tasks/install-system-release.yml`. __It's only responsible for installing already existing "system" Helm charts from the "system" repository.__
```

File: docs/home/howto/HELM.md [hunk 4]
```diff
@@ -1,78 +1,67 @@
...
+This helper task expects such parameters/facts:
 
+```yaml
+- set_fact:
+    helm_chart_name: <string>
+    helm_chart_values: <map>
+    helm_release_name: <string>
+```
 
+- `helm_chart_values` it's a standard yaml map, values defined there replace default config of the chart (`values.yaml`).
```
Comment (to-bar) [inline docs/home/howto/HELM.md:24]:
```suggestion
- `helm_chart_values` is a standard yaml map, values defined there replace default config of the chart (`values.yaml`).
```

File: docs/home/howto/HELM.md [hunk 5]
```diff
@@ -1,78 +1,67 @@
...
 
-Single Helm installation task is responsible for installing already existing charts from repository.
+Our standard practice is to place those values inside the `specification` document of the role that deploys the helm release in Kubernetes.
```
Comment (to-bar) [inline docs/home/howto/HELM.md:26]:
```suggestion
Our standard practice is to place those values inside the `specification` document of the role that deploys the Helm release in Kubernetes.
```

File: docs/home/howto/HELM.md [hunk 6]
```diff
@@ -1,78 +1,67 @@
...
-    disable_helm_chart: "{{ _disable_helm_chart }}"
-    helm_chart_name: "{{ _helm_chart_name }}"
-    helm_chart_values:  "{{ _helm_chart_values }}"
+    helm_chart_name: "{{ specification.helm_chart_name }}"
+    helm_release_name: "{{ specification.helm_release_name }}"
+    helm_chart_values: "{{ specification.helm_chart_values }}"
 ```
 
-Example config:
+__By default all installed helm "system" releases are deployed inside the `epi-charts` namespace in Kubernetes.__
```
Comment (to-bar) [inline docs/home/howto/HELM.md:55]:
```suggestion
__By default all installed "system" Helm releases are deployed inside the `epi-charts` namespace in Kubernetes.__
```

File: docs/home/howto/HELM.md [hunk 7]
```diff
@@ -1,78 +1,67 @@
...
+
+## Removing "system" helm releases
```
Comment (to-bar) [inline docs/home/howto/HELM.md:57]:
```suggestion
## Uninstalling "system" Helm releases
```

File: docs/home/howto/HELM.md [hunk 8]
```diff
@@ -1,78 +1,67 @@
...
+
+To delete helm releases `roles/helm/tasks/delete-system-release.yml` can be used. For example:
```
Comment (to-bar) [inline docs/home/howto/HELM.md:59]:
```suggestion
To uninstall Helm release `roles/helm/tasks/delete-system-release.yml` can be used. For example:
```

**Review by to-bar — CHANGES_REQUESTED** [review]

File: core/src/epicli/data/common/ansible/playbooks/roles/helm_charts/files/system/epi-dummy/Chart.yaml [hunk 9]
```diff
@@ -0,0 +1,23 @@
+apiVersion: v2
+name: epi-dummy
+description: A Helm chart for Kubernetes
```
Comment (sk4zuzu) [inline core/src/epicli/data/common/ansible/playbooks/roles/helm_charts/files/system/epi-dummy/Chart.yaml:3]:
```suggestion
description: A dummy haproxy-based chart for testing purposes
```

File: core/src/epicli/data/common/ansible/playbooks/roles/helm_charts/files/system/epi-dummy/templates/tests/test-connection.yaml [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  name: "{{ include "epi-dummy.fullname" . }}-test-connection"
+  labels:
+    {{- include "epi-dummy.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": test-success
+spec:
+  containers:
+    - name: wget
+      image: busybox
```
Comment (sk4zuzu) [inline core/src/epicli/data/common/ansible/playbooks/roles/helm_charts/files/system/epi-dummy/templates/tests/test-connection.yaml:12]:
```suggestion
      # NOTE: We do not have busybox image in epirepo, so this test will not work in offline mode.
      image: busybox
```

File: core/src/epicli/data/common/ansible/playbooks/roles/helm_charts/files/system/epi-dummy/Chart.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (sk4zuzu) [reply core/src/epicli/data/common/ansible/playbooks/roles/helm_charts/files/system/epi-dummy/Chart.yaml]: Comment added in the code. :+1::innocent:

File: core/src/epicli/data/common/ansible/playbooks/roles/helm_charts/files/system/epi-dummy/templates/tests/test-connection.yaml (hunk 10 above)
[the remark this answers is not in the document]
Comment (sk4zuzu) [reply core/src/epicli/data/common/ansible/playbooks/roles/helm_charts/files/system/epi-dummy/templates/tests/test-connection.yaml:13]: Comment added in the code. :+1::innocent:

**Review by to-bar — APPROVED** [review]

**Review by erzetpe — APPROVED** [review]

**Pull request merged**